src/input/mod.rs

```diff
@@ -0,0 +1,62 @@
+use bevy::prelude::*;
+
+use crate::entities::character::init::{CurrentAction, ProcessNewActionEvent};
+use crate::{App, KeyCode, SceneState};
+
+pub struct InputPlugin;
+
+impl Plugin for InputPlugin {
+    fn build(&self, app: &mut App) {
+        app.add_system_set(
+            SystemSet::on_update(SceneState::ExplorationMode).with_system(character_movement_input),
+        );
+    }
+}
+
+const DEFAULT_VELOCITY: f32 = 100.0;
+
+fn character_movement_input(
+    mut ev_update: EventWriter<ProcessNewActionEvent>,
+    keyboard_input: Res<Input<KeyCode>>,
+    _time: Res<Time>,
+) {
+    let mut velocity: Vec3 = Vec3::ZERO;
+    let mut direction_switched = false;
+    if keyboard_input.pressed(KeyCode::Z) {
+        velocity.y += DEFAULT_VELOCITY;
+        ev_update.send(ProcessNewActionEvent {
+            new_action: CurrentAction::Up,
+        });
+        direction_switched = true;
+    }
+    if keyboard_input.pressed(KeyCode::S) {
+        velocity.y -= DEFAULT_VELOCITY;
+
+        ev_update.send(ProcessNewActionEvent {
+            new_action: CurrentAction::Down,
+        });
+        direction_switched = true;
+    }
+    if keyboard_input.pressed(KeyCode::Q) {
+        velocity.y -= DEFAULT_VELOCITY;
+
+        ev_update.send(ProcessNewActionEvent {
+            new_action: CurrentAction::Left,
+        });
+        direction_switched = true;
+    }
+    if keyboard_input.pressed(KeyCode::D) {
+        velocity.y -= DEFAULT_VELOCITY;
+
+        ev_update.send(ProcessNewActionEvent {
+            new_action: CurrentAction::Right,
+        });
+        direction_switched = true;
+    }
+
+    if !direction_switched {
```

Comment:
```suggestion
    if velocity == Vec3::ZERO {
```
and then, delete this direction_switched parameter

Reply:
leftover from previous code, velocity has no meaning in input. I deleted this field.

Reply:
Also an issue when pressing 2+ keys at the same time. Not sure how to solve it

Reply:
(from a gameplay perspective)